test/integration/dnsrecord/dnsrecord_test.go

```diff
@@ -0,0 +1,294 @@
+		Expect(awsinstall.AddToScheme(mgr.GetScheme())).To(Succeed())
+
+		Expect(dnsrecordctrl.AddToManager(mgr)).To(Succeed())
+
+		var mgrContext context.Context
+		mgrContext, mgrCancel = context.WithCancel(ctx)
+
+		By("starting manager")
+		go func() {
+			err := mgr.Start(mgrContext)
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		// test client should be uncached and independent from the tested manager
+		c, err = client.New(cfg, client.Options{
+			Scheme: mgr.GetScheme(),
+			Mapper: mgr.GetRESTMapper(),
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(c).NotTo(BeNil())
+
+		flag.Parse()
+		validateFlags()
+
+		awsClient, err = awsclient.NewClient(*accessKeyID, *secretAccessKey, aws.DefaultDNSRegion)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterSuite(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+
+		By("running cleanup actions")
+		framework.RunCleanupActions()
+
+		By("stopping test environment")
+		Expect(testEnv.Stop()).To(Succeed())
+	})
+
+	It("should successfully create and delete", func() {
+		By("setting up shoot environment")
+		Expect(c.Create(ctx, namespace)).To(Succeed())
+		Expect(c.Create(ctx, secret)).To(Succeed())
+		Expect(c.Create(ctx, cluster)).To(Succeed())
+		framework.AddCleanupAction(func() {
+			By("tearing down shoot environment")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+			Expect(client.IgnoreNotFound(c.Delete(ctx, secret))).To(Succeed())
+			Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+		})
+
+		By("creating dnsrecord")
+		dns := newDNSRecord(testName)
+		Expect(c.Create(ctx, dns)).To(Succeed())
+		framework.AddCleanupAction(func() {
+			By("deleting dnsrecord")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, dns))).To(Succeed())
+
+			By("waiting until dnsrecord is deleted")
+			Expect(extensions.WaitUntilExtensionObjectDeleted(
+				ctx,
+				c,
+				logger,
+				dns.DeepCopy(),
+				extensionsv1alpha1.DNSRecordResource,
+				10*time.Second,
+				5*time.Minute,
+			)).To(Succeed())
+
+			By("verifying dnsrecord deletion")
+			verifyDeletion(ctx, awsClient, dns)
+		})
+
+		By("waiting until dnsrecord is reconciled")
+		Expect(extensions.WaitUntilExtensionObjectReady(
+			ctx,
+			c,
+			logger,
+			dns,
+			extensionsv1alpha1.DNSRecordResource,
+			10*time.Second,
+			30*time.Second,
+			5*time.Minute,
+			nil,
+		)).To(Succeed())
+
+		By("getting dnsrecord")
+		Expect(c.Get(ctx, client.ObjectKey{Namespace: dns.Namespace, Name: dns.Name}, dns)).To(Succeed())
+
+		By("verifying that the dnsrecord's status contains the zone")
+		Expect(dns.Status.Zone).To(PointTo(Equal(*zoneID)))
+
+		By("verifying dnsrecord creation")
+		verifyCreation(ctx, awsClient, dns)
+	})
+})
+
+func newDNSRecord(namespace string) *extensionsv1alpha1.DNSRecord {
+	return &extensionsv1alpha1.DNSRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dnsrecord",
+			Namespace: namespace,
+		},
+		Spec: extensionsv1alpha1.DNSRecordSpec{
```

Comment:
Should we add another test case with `spec.zone` set?

Reply:
This is still not finished (so `Draft`), I plan to add not one, but multiple "interesting" test cases.

Reply:
Added a few more test cases that cover this and other interesting situations.